### PR TITLE
OFFENCES-84 Add delta sync functionality to NOMIS, add toggling capability to turn features on/off through api calls, add schedule for full NOMIS sync

### PR DESCRIPTION
**Pull request by yasinmustafa**
- add a schedule to run full sync to NOMIS (every hour) - however will only run if the the toggle is enabled
- Three feature toggles added
1. FULL_SYNC_NOMIS
2. DELTA_SYNC_NOMIS (runs after the sync from SDRS - as part of the same job))
3. SYNC_SDRS

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/resource/OffenceController.kt [hunk 1]
```diff
@@ -48,17 +45,6 @@ class OffenceController(
     return offenceService.findLoadResults()
   }
 
-  @PostMapping(value = ["/full-sync-nomis"])
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/resource/OffenceController.kt:51]: the full sync to NOMIS now runs on a schedule if the toggle feature is enabled.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/OffenceService.kt [hunk 2]
```diff
@@ -30,15 +33,20 @@ class OffenceService(
     return sdrsLoadResultRepository.findAllByOrderByAlphaCharAsc().map { transform(it) }
   }
 
+  @Scheduled(cron = "0 0 */1 * * *")
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/OffenceService.kt:36]: scheduled to run once every hour

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/resource/OffencesControllerIntTest.kt [hunk 3]
```diff
@@ -103,118 +100,4 @@ class OffencesControllerIntTest : IntegrationTestBase() {
         )
     }
   }
-
```
Comment (yasinmustafa) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/resource/OffencesControllerIntTest.kt:106]: this endpoint no longer exists - the full sync to nomis  is no longer directly triggered by an api point and insread runs on a schedule protected by a feature toggle

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@ spring:
   application:
     name: manage-offences-api
   codec:
-    max-in-memory-size: 10MB
+    max-in-memory-size: 15MB
```
Comment (yasinmustafa) [inline src/main/resources/application.yml:9]: this was changed because some of the responses from SDRS were going over 10MB

**Review by joelstobart-moj — APPROVED** [review]
LGTM

**Pull request merged**